Gems/OpenXRVk/Code/Source/XRCameraMovementComponent.cpp

```diff
@@ -0,0 +1,167 @@
+    {
+    }
+
+    void XRCameraMovementComponent::Activate()
+    {
+        AzFramework::InputChannelEventListener::Connect();
+        AZ::TickBus::Handler::BusConnect();
+    }
+
+    void XRCameraMovementComponent::Deactivate()
+    {
+        AZ::TickBus::Handler::BusDisconnect();
+        AzFramework::InputChannelEventListener::Disconnect();
+    }
+
+    void XRCameraMovementComponent::OnTick(float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint timePoint)
+    {
+        AZ::Transform cameraTransform = GetCameraTransformFromCurrentView();
+
+        // Update movement...
+        const float moveSpeed = m_moveSpeed * deltaTime;
+        const AZ::Vector3 movementVec = (cameraTransform.GetBasisX() * m_movement.GetX())
+            + (cameraTransform.GetBasisY() * m_movement.GetY())
+            + (AZ::Vector3{0.f, 0.f, 1.f} * m_movement.GetZ()); // use a fixed UP for the Z direction
+        const AZ::Vector3 newPosition{ (cameraTransform.GetTranslation() + (movementVec * moveSpeed)) };
+        cameraTransform.SetTranslation(newPosition);
+
+        AZ::TransformBus::Event(GetEntityId(), &AZ::TransformBus::Events::SetWorldTM, cameraTransform);
+    }
+
+    bool XRCameraMovementComponent::OnInputChannelEventFiltered([[maybe_unused]] const AzFramework::InputChannel& inputChannel)
+    {
+        const auto& deviceId = inputChannel.GetInputDevice().GetInputDeviceId();
+        if (AzFramework::InputDeviceXRController::IsXRControllerDevice(deviceId))
+        {
+            OnXRControllerEvent(inputChannel);
+        }
+        return false;
+    }
+
+    void XRCameraMovementComponent::OnXRControllerEvent([[maybe_unused]] const AzFramework::InputChannel& inputChannel)
+    {
+        const auto& channelId = inputChannel.GetInputChannelId();
+
+        // This currently uses specific xr controller channels to drive the movement.  Future iterations might
+        // use a higher-level concepts like InputMappings and InputContexts to generalize to additional
+        // input devices.
+
+        // Left thumb-stick X/Y move the camera
+        if (channelId == AzFramework::InputDeviceXRController::ThumbStickAxis1D::LX)
+        {
+            m_movement.SetX(inputChannel.GetValue() * m_movementSensitivity);
+        }
+        if (channelId == AzFramework::InputDeviceXRController::ThumbStickAxis1D::LY)
+        {
+            m_movement.SetY(inputChannel.GetValue() * m_movementSensitivity);
+        }
+
+        // A/B buttons update the height in Z of the camera
+        if (channelId == AzFramework::InputDeviceXRController::Button::A)
+        {   // down
+            m_movement.SetZ(-inputChannel.GetValue() * m_movementSensitivity);
+        }
+        if (channelId == AzFramework::InputDeviceXRController::Button::B)
+        {   // up
+            m_movement.SetZ(inputChannel.GetValue() * m_movementSensitivity);
+        }
+    }
+
+    // static
```

Comment:
left over comment? please remove `// static`.

Reply:
Removed.  Although, I think this function could just be a static function in the cpp, doesn't need to be a class static.  I'll update it.